lib/src/components/NxToggle/__tests__/NxToggleRTL.test.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+import '@testing-library/jest-dom';
+
+import { render, screen, fireEvent } from '@testing-library/react';
+
+import NxToggle, { Props } from '../NxToggle';
+
+describe('NxToggle', function() {
+  const simpleProps: Props = {
+    inputId: 'toggle-id',
+    isChecked: false,
+    onChange: () => {},
+    disabled: undefined,
+    children: 'Enables whales'
+  };
+
+  it('renders a <label> containing a toggle <input> and toggle control', function() {
+    render(<NxToggle {...simpleProps}></NxToggle>);
+
+    const checkbox = screen.getByRole('switch');
+
+    expect(checkbox).toHaveAttribute('type', 'checkbox');
+    expect(checkbox).toHaveAttribute('id', 'toggle-id');
+    expect(checkbox).toHaveClass('nx-toggle__input');
+
+  });
+
+  it('adds classes specified with the className prop', function() {
+    const { container } = render(<NxToggle {...simpleProps} className="foo"></NxToggle>);
+
+    const label = container.querySelector('label');
+
+    expect(label).toHaveClass('foo');
+    expect(label).toHaveClass('nx-toggle');
+  });
+
+  it('calls its onChange prop when the input fires a change event', function() {
```

Comment:
This test is no longer testing the change event directly, it's testing the click event.  I guess an additional test would be needed to test keyboard activation.  I wonder if there are patterns for that, so we don't have to encode all the ways a control can be activated into each test suite?

Reply:
> This test is no longer testing the change event directly, it's testing the click event. I guess an additional test would be needed to test keyboard activation. I wonder if there are patterns for that, so we don't have to encode all the ways a control can be activated into each test suite?

Turns out there is `fireEvent.change` but it doesn't work with checkboxes.
https://github.com/testing-library/react-testing-library/issues/156

Most of the discussions on the topic is to use fire the click event.
I can't seem to find any other pattern to do both so it does look like we'll have to test keyboard activation separately.

Reply:
It's not clear to me whether it makes a difference in this case, but some of the RTL docs refer to this additional library that sounds like it's somewhat relevant and is something we should be aware of: https://github.com/testing-library/user-event